File: src/main/java/no/nav/sosialhjelp/soknad/consumer/pdl/geografisktilknytning/dto/GeografiskTilknytningDto.java
//package no.nav.sosialhjelp.soknad.consumer.pdl.geografisktilknytning.dto;
//
//import com.fasterxml.jackson.annotation.JsonCreator;
//import com.fasterxml.jackson.annotation.JsonProperty;
//
//public class GeografiskTilknytningDto {
//
//    private final GtType gtType;
//    private final String gtKommune;
//    private final String gtBydel;
//    private final String gtLand;
//
//    @JsonCreator
//    public GeografiskTilknytningDto(
//            @JsonProperty("gtType") GtType gtType,
//            @JsonProperty("gtKommune") String gtKommune,
//            @JsonProperty("gtBydel") String gtBydel,
//            @JsonProperty("gtLand") String gtLand
//    ) {
//        this.gtType = gtType;
//        this.gtKommune = gtKommune;
//        this.gtBydel = gtBydel;
//        this.gtLand = gtLand;
//    }
//
//    public GtType getGtType() {
//        return gtType;
//    }
//
//    public String getGtKommune() {
//        return gtKommune;
//    }
//
//    public String getGtBydel() {
//        return gtBydel;
//    }
//
//    public String getGtLand() {
//        return gtLand;
//    }
//}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/navenhet/gt/dto/GeografiskTilknytningDto.kt
package no.nav.sosialhjelp.soknad.navenhet.gt.dto

data class HentGeografiskTilknytning(
    val hentGeografiskTilknytning: GeografiskTilknytningDto?
)

data class GeografiskTilknytningDto(
    val gtType: GtType,
    val gtKommune: String?,
    val gtBydel: String?,
    val gtLand: String?
)

/**
 * Sjekker om en GT er norsk, og dermed hvorvidt en fire- eller seks-sifret GT-kode kan utledes
 */
fun GeografiskTilknytningDto.erNorsk(): Boolean = gtType == GtType.KOMMUNE || gtType == GtType.BYDEL

/**
 * Henter fire- eller sekssifret GT-kode fra GeografiskTilknytningDto.
 * Returnerer gtKommune hvis gtType er KOMMUNE, gtBydel hvis gtType er BYDEL, ellers null
 */
fun GeografiskTilknytningDto.toGtStringOrThrow(): String = when (gtType) {
    GtType.BYDEL -> gtBydel ?: error("gtType er BYDEL men gtBydel er null")
    GtType.KOMMUNE -> gtKommune ?: error("gtType er KOMMUNE men gtKommune er null")
    else -> error("gtType er hverken BYDEL eller KOMMUNE")
}

enum class GtType {
    BYDEL, KOMMUNE, UDEFINERT, UTLAND
}
